script.js

```diff
@@ -3,29 +3,62 @@ var mainPalette = {
   buttonContainer: document.querySelector('.button-container'),
   savedPaletteContainer: document.querySelector('.saved-palettes-container'),
   currentPalette: '',
+  savedPalettes: [],
+
+  newPaletteButton() {
+      this.currentPalette.refresh();
+      this.displayMain();
+  },
+
+  savePaletteButton() {
+    if (this.savedPalettes.length === 8) {
+      this.savedPalettes.pop();
+    }
+    this.savedPalettes.unshift(this.currentPalette);
+    this.resetLocks();
+    this.createNew();
+    this.displaySaved();
+  },
 
   createNew() {
       this.currentPalette = new Palette();
-      this.display()
+      this.displayMain();
   },
 
-  newPaletteButton() {
-      this.currentPalette.refresh();
-      this.display()
+  resetLocks() {
+    this.currentPalette.palette.forEach((color) => {
```

Comment:
Nice use of forEach! This will be something they will teach you in Mod2, so make sure you understand how to explain what forEach does. I'm willing to bet you will get feedback on using it.